nifty/pipeline/nifsUtils.py

```diff
@@ -1460,5 +1462,151 @@ def hasBPMExt(self, file):
+                for i in range(len(cals.arcdarks)):
+                    hdul['PRIMARY'].header['DCMB'+str(i+1)] = (os.path.split(cals.arcdarks[i])[1], self.keywordDict['DCMB'])
+                hdul.flush()
+            if not self.hasBPMExt(cals.arc_file):
+                iraf.imcopy(cals.bpm_file, cals.arc_file+"[BPM,type=mask,append]")
+        except Exception:
+            logging.error("Problem tagging {}.".format(cals.arc_file))
+
+    def tagRonchi(self, cals):
+        try:
+            with fits.open(cals.ronchi_file, mode="update") as hdul:
+                if "-RONCHI" not in hdul['PRIMARY'].header['DATALAB']:
+                    hdul['PRIMARY'].header['DATALAB'] = hdul['PRIMARY'].header['DATALAB']+"-RONCHI"
+                if ".fits" not in hdul['PRIMARY'].header['FLATIMAG']:
+                    hdul['PRIMARY'].header['FLATIMAG'] = hdul['PRIMARY'].header['FLATIMAG']+".fits"
+                if ".fits" not in hdul['PRIMARY'].header['DARKIMAG']:
+                    hdul['PRIMARY'].header['DARKIMAG'] = hdul['PRIMARY'].header['DARKIMAG']+".fits"
+                hdul['PRIMARY'].header['SHFILE'] = (os.path.split(cals.shift_file)[1], self.keywordDict['SHFILE'])
+                hdul['PRIMARY'].header['FCOMBINE'] = (len(cals.ronchis), self.keywordDict['FCOMBINE'])
+                for i in range(len(cals.ronchis)):
+                    hdul['PRIMARY'].header['FCMB'+str(i+1)] = (os.path.split(cals.ronchis[i])[1], self.keywordDict['FCMB'])
+                hdul['PRIMARY'].header['DCOMBINE'] = (len(cals.flatdarks), self.keywordDict['DCOMBINE'])
+                for i in range(len(cals.flatdarks)):
+                    hdul['PRIMARY'].header['DCMB'+str(i+1)] = (os.path.split(cals.flatdarks[i])[1], self.keywordDict['DCMB'])
+                hdul.flush()
+            if not self.hasBPMExt(cals.ronchi_file):
+                iraf.imcopy(cals.bpm_file, cals.ronchi_file+"[BPM,type=mask,append]")
+        except Exception:
+            logging.error("Problem tagging {}.".format(cals.ronchi_file))
+
+
+    def tagShift(self, cals):
+        pass
+
+    def hasBPMExt(self, file):
+        hdulist = astropy.io.fits.open(file)
+        return any([x.name == "BPM" for x in hdulist])
+
+    def parseList(self, listpath):
+        """
+        Tries to open a list in listpath, asserts it's non-empty, and well-formed.
+        """
+        with open(listpath) as f:
+            lines = f.readlines()
+
+        lines = [x.rstrip('\n') + '.fits' for x in lines]
+
+        assert len(lines) > 0
+
+        assert all(["N2" in x for x in lines])
+
+        basePath = os.path.split(listpath)[0]
+
+        lines = [os.path.join(basePath, x) for x in lines]
+
+        return lines
+
+    def getCalFile(self, calpath):
+        try:
+            calfile = glob.glob(calpath)[0]
+        except IndexError as e:
+            logging.error("Calibration file {} not found.".format(calpath))
+            raise CalibrationsError()
```

Comment:
Do you intend to replace the well-known FileNotFoundError with a custom CalibrationsError? If the error raised intentionally changes downstream processing, that might be ok, but otherwise you've added un-necessary code and complexity.

I don't know off the top of my head if os.path.exists is a cheaper call than glob.glob, but using glob is an interesting way of finding out if a path exists.